lib/shared/assert.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @fileoverview Assertion utilities equivalent to the Node.js node:asserts module.
+ * @author Josh Goldberg
+ */
+
+"use strict";
+
+/**
+ * Throws an error if the input is not truthy.
+ * @param {unknown} value The input that is checked for being truthy.
+ * @param {string} message Message to throw if the input is not truthy.
+ * @returns {void}
+ * @throws {Error} When the condition is not truthy.
+ */
+function ok(value, message) {
+    if (!value) {
+        throw new Error(message);
+    }
+}
+
+
+module.exports = {
+    ok,
+
+    /*
+     * Intentionally mirroring the node:assert interface, instead of normal Knip CommonJS patterns.
+     * https://knip.dev/guides/working-with-commonjs
+     */
+    __esModule: true
+};
```

Comment:
Thoughts about simplifying this to export a function? I think all we need in production code is a simple `assert(value, message)` as a shorthand for `if (!value) { throw new Error(message); }`.

Reply:
I slightly prefer it as-is, since `node:assert` is a pretty common standard to build against. I don't like how a lot of projects have ended up with their own subtly different ad-hoc reimplementations of it. I've found it confusing to swap between projects, and inconvenient to migrate to more standard solutions. But I wouldn't push back hard if there's otherwise consensus to streamline.

Reply:
I'm not sure if the goal is to mimic the `node:assert` API, but even if it is, it supports `assert(value, message)`:

https://nodejs.org/api/assert.html#assertvalue-message

Reply:
Good point, done!